File: Lakshay Garg/Leetcode/Sliding Window/Alternating Groups II.cpp
class Solution {
public:
    int numberOfAlternatingGroups(vector<int>& arr, int k) {
        int n=arr.size(),count=1,ans=0;
        for(int i=1;i<n+k-1;i++){
            if(arr[i%n]!=arr[(i-1)%n]) count++;
            else count=1;
            if(count>=k) ans++;
        }
        return ans;
    }
};

File: Shantanu_Mamgain/Sliding-Window-Leetcode/Alternating Groups II.java
class Solution {
    public int numberOfAlternatingGroups(int[] colors, int k) {
        int start = 0;
        int n = colors.length;
        int count = 0;
        
        for(int end = 1; end < n+k-1; end++){
            if(colors[end % n] != colors[(end+1) % n] ){
                if(end-start+1 >= k ){
                    count ++;
                }
            }
            else{
                start = end;
            }
        }
        return count;
    }
}
